### PR TITLE
#Assignment 3

### PR DESCRIPTION
**Pull request by OlivierPaulo**
Including integration and units tests in `test_cleaning.py`

Some updates following refactoring:
- use of `pd.DataFrame.pipe(...)`
- update python version of reference in the dev
- add new argument to handle file_name
- use of `*args` and `**kwargs` in `main` function

Adding black reformating into the CI.

### CODE REVIEW

File: life_expectancy/tests/test_data.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import pandas as pd
+
+from life_expectancy.data import load_data, save_data
+from . import OUTPUT_DIR, FIXTURES_DIR
+
+
+def test_load_data():
+    """Run unit test of function `load_data`"""
+    pd.testing.assert_frame_equal(
+        load_data(f"{FIXTURES_DIR}/eu_life_expectancy_raw_fixture.tsv"),
```
Comment (antoniomcoelho) [inline life_expectancy/tests/test_data.py]: One small thing: for better reading, I would suggest to create variables (for example, actual_data and expected_data) and then use pd.testing.assert_frame_equal(actual_data, expected_data). I think it allows for a better understanding of the code.

**Review by antoniomcoelho — APPROVED** [review]
I think the work done was good. I just left a small note (maybe its more about personal preferences, so its not important).

File: life_expectancy/tests/test_data.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        f"{FIXTURES_DIR}/eu_life_expectancy_raw_fixture.tsv", sep="\t"
+    )
+    pd.testing.assert_frame_equal(actual_data, expected_data)
+
+
+def test_save_data(pt_life_expectancy_expected):
+    """Run unit test of function `save_data`"""
+    with patch.object(pt_life_expectancy_expected, "to_csv") as to_csv_mock:
+        save_data(pt_life_expectancy_expected)
+        to_csv_mock.assert_called_with(
```
Comment (MrCordeiro) [inline life_expectancy/tests/test_data.py]:
```suggestion
    with patch.object(pt_life_expectancy_expected, "to_csv") as to_csv_mock:
        # to_csv_mock.return_value = MY_FIXTURE
        to_csv_mock.side_effect = print("Hello mommy!)
        save_data(pt_life_expectancy_expected)
        to_csv_mock.assert_called_with(
```

Comment (MrCordeiro) [reply life_expectancy/tests/test_data.py]: capfd ou capsys

File: life_expectancy/tests/test_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+"""Tests for the pipeline module"""
+import pandas as pd
+
+from life_expectancy.pipeline import main
+from . import FIXTURES_DIR
+
+
+def test_main(pt_life_expectancy_expected):
```
Comment (MrCordeiro) [inline life_expectancy/tests/test_pipeline.py]:
```suggestion
@patch("life_expectancy.pipeline.save_data")
def test_main(mock, pt_life_expectancy_expected):
    mock.side_effect = XXX
    mock.assert_called_with(ARGS_TO_BE_TESTED)
```

**Pull request merged**